### DISCUSSION TITLE
Ignoring "engines" field for Node.js projects

### DISCUSSION THREAD

**Discussion by binyamin** [renovatebot/renovate#13521] [Request Help] [answered]
I'd like renovate to ignore the `engines` depType for Node.js. I could only find an option to _match_ depTypes, ie. by whitelisting them.

According to [NPM's documentation on the package.json][engines], the `engines` field is meant for <q cite="https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines">the version of node that your stuff works on</q>. And, NPM will produce warnings if the user's environment doesn't match the required node version.

In other words, a package which breaks on v10 might have it set to v12. If set to v16, users with v12 or v14 [may _not be able_ to install it][engine-strict].

[engines]: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines
[engine-strict]: https://docs.npmjs.com/cli/v8/using-npm/config#engine-strict

**Comment by viceice** [comment, marked as answer]
```json
    {
      "description": "Ignore nodejs",
      "matchPackageNames": ["node"],
      "matchManagers": ["npm"],
      "matchDepTypes": [ "engines" ],
      "dependencyDashboardApproval": true
    },
```

So any node update needs approval, you can see them on dashboard.

**Reply by binyamin** [reply]
I don't really want PRs at all.

Also, why doesn't renovate default to this behavior? That's not what the field is meant for. For example, I have a package ([@binyamin/data-cache](https://www.npmjs.com/package/@binyamin/data-cache)) with `engines.node` set to `>=v12`. That tells NPM that the package breaks on v10. If I set it to v16, users with v12 or v14 [_may not be able to install it_](https://docs.npmjs.com/cli/v8/using-npm/config#engine-strict).

**Reply by viceice** [reply]
use something like this?

https://github.com/renovatebot/renovate/blob/156041e6a98b423b846c032de2920fcc8e2fa661/package.json#L123

**Reply by viceice** [reply]
renovate tries to updates all possible, so you need to disable manually. **We don't recommend to disable updates!**

```json
    {
      "description": "Ignore nodejs",
      "matchPackageNames": ["node"],
      "matchManagers": ["npm"],
      "matchDepTypes": [ "engines" ],
      "enabled": false
    },
```

**Reply by binyamin** [reply]
thanks. I'll try `"node": "^14.15.0 || >=16.13.0"`.

**Reply by viceice** [reply]
I think your issue is your config, you're grouping all updates and use bump reange strategy here https://github.com/binyamin/.github/blob/master/renovate-config.json

so renovate will always try to bump your engines. so you should use a package rule to use default range strategy for engines, so renovate will only ever propose an update for security or out of range updates.

**Reply by binyamin** [reply]
Speaking as a maintainer, I can appreciate that you're helping one of many strangers, with a bug that's really just user-error. This selfless behavior is the backbone of open-source. Unfortunately, I don't see it very often. Thank you.
